extensions/assets-manager/AssetsManagerEx.cpp

```diff
@@ -476,11 +498,14 @@ void AssetsManagerEx::parseVersion()
         }
         else
         {
+            // Save the old update entry type since its value may be changed in event listener.
+            UpdateEntry oldUpdateEntry = _updateEntry;
+
             _updateState = State::NEED_UPDATE;
             dispatchUpdateEvent(EventAssetsManagerEx::EventCode::NEW_VERSION_FOUND);
 
             // Wait to update so continue the process
-            if (_waitToUpdate)
+            if (oldUpdateEntry == UpdateEntry::DO_UPDATE)
```

Comment:
This logic is from the original code. I don't wanna change big logic of old code.
My thought is: if developers didn't invoke `checkUpdate`, he/she just invoke `update` method, then `oldUpdateEntry` is `DO_UPDATE` here, need to `continue` the following steps like downloading manifest, assets ..., otherwise, the flow may be stopped.